amd_tressfx_vulkan/src/TressFXOpaqueVulkan.cpp

```diff
@@ -0,0 +1,228 @@
+//
+// Copyright (c) 2016 Advanced Micro Devices, Inc. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#include "TressFXOpaqueVulkan.h"
+
+namespace AMD
+{
+void TressFX_OpaqueDesc::Initialize(TressFX_Desc &desc, VkImageView depthTexture,
+                                    VkImageView colorTexture,
+                                    VkCommandBuffer commandBuffer,
+                                    VkDeviceMemory scratchMemory, VkBuffer scratchBuffer,
+                                    size_t &offsetInScratchBuffer,
+                                    uint32_t cpu_memory_index, uint32_t gpu_memory_index)
+{
+    if (initialized == false)
```

Comment:
Please refactor this such that the `TressFX_OpaqueDesc` construct sets the ref count to 0, and remove the `initialized` variable which doesn't serve a purpose any more then.